### DISCUSSION TITLE
Reorder crashes when using views --> Undefined array key "entry"

### DISCUSSION THREAD

**Discussion by hanneshoepfnerio** [statamic/cms#9304] [Troubleshooting]
Hello, I'm experiencing some pretty heavy error I guess.
I have a collection which is orderable. In the backend UI I can reorder – no problem. But I have a view which filters some elements of that collection. When I use this view when reordering, the whole collection crashes and cannot be displayed anymore.
Here is the error

```
Undefined array key "entry"
######vendor/statamic/cms/src/Structures/CollectionTreeDiff.php#40
ErrorException
    private function flatten($arr)
    {
        return collect($arr)->mapWithKeys(function ($item, $i) {
            $results = [$item['entry'] => [
                'path' => $item['path'],
                'index' => $i,
            ]];
```
Would be great to fix this since the reorder feature in combination with view filters bring great functionality.
For example if you are managing event in that collection, but they have a taxonomy connected and you want to reorder only the elements with that taxonomy. Otherwise it's a pain and I would always need to use a entry relation on another page as proxy for that.

**Comment by duncanmcclean** [comment]
> But I have a view which filters some elements of that collection. When I use this view when reordering, the whole collection crashes and cannot be displayed anymore.

I'm not quite sure I understand what you mean by "I have a view which filters some elements of that collection". Is this something custom you've built?

**Reply by hanneshoepfnerio** [reply]
No, standard functionality. When in collection "view" - you can access different fields to filter and save this as view.
This "view" / preset of filter, is available when reordering.
I wrote a custom filter and when using it all items are visible on default and all filters are disabled on reorder.
So it seems to be a "bug" in being able to access the view while reordering, BUT I would very much welcome this functionality, because it would really help in a lot of situations that I have had in the past.
Maybe you could include it while fixing?

Best

**Reply by duncanmcclean** [reply]
Oh sorry, I see what you mean now.

I think that's *probably* the correct behaviour. If you have 100 entries and a view which displays just 2 of them, and you re-order the two, there's no way for us to know where in the 100, those 2 entries should sit. 

I'm not sure why it's erroring though, that's a bug. Can you open a bug report containing the full stack trace of the error?

**Reply by hanneshoepfnerio** [reply]
Hi – true, makes sense. I also wouldn't know how to implement this, but it would be great ;)
I'll open a bug report, should I mention you?

**Reply by duncanmcclean** [reply]
No, just mention this discussion.
